scheduler/feasible.go

```diff
@@ -850,35 +854,73 @@ func checkAttributeAffinity(ctx Context, operand string, lVal, rVal *psstructs.A
 	return checkAttributeConstraint(ctx, operand, lVal, rVal, lFound, rFound)
 }
 
-// checkLexicalOrder is used to check for lexical ordering
-func checkLexicalOrder(op string, lVal, rVal interface{}) bool {
-	// Ensure the values are strings
-	lStr, ok := lVal.(string)
-	if !ok {
+// checkOrder returns the result of (lVal operand rVal). The comparison is
+// done as integers if possible, or floats if possible, and lexically otherwise.
+func checkOrder(operand string, lVal, rVal any) bool {
+	left, leftOK := lVal.(string)
+	right, rightOK := rVal.(string)
+	if !leftOK || !rightOK {
 		return false
 	}
-	rStr, ok := rVal.(string)
-	if !ok {
-		return false
+	if result, ok := checkIntegralOrder(operand, left, right); ok {
+		return result
+	}
+	if result, ok := checkFloatOrder(operand, left, right); ok {
+		return result
 	}
+	return checkLexicalOrder(operand, left, right)
+}
+
+// checkIntegralOrder compares lVal and rVal as integers if possible, or false otherwise.
+func checkIntegralOrder(op, lVal, rVal string) (bool, bool) {
+	left, lErr := strconv.Atoi(lVal)
```

Comment:
It _probably_ doesn't matter, but this restricts the integer size to a machine-sized int instead of a int64, which means on the small number of 32-bit platforms we support we could end up with different behavior. `strconv.ParseInt(lVal, 10, 64)` would be more specific. (But also, do we care?)

Reply:
I could see this being an issue if someone started using this comparison for `unique.storage.bytesfree`; I'll add the fix